### ISSUE TITLE
Document registry deletes are not synced

### ISSUE THREAD

**Issue by lache-melvin** [msupply-foundation/open-msupply#5820]
## What went wrong? 😲

Found while setting up client demo:

You can delete document registry entries (i.e. config for an encounter) from OG, but this is not synced to Open mSupply.

If you make a mistake, so delete and reupload your config, you now have duplicates in OMS. Only way to resolve is to delete directly from DB.

## Expected behaviour 🤔

Deletes of document registry items are synced from OG. I think need to then soft delete in OMS so reference to document_type for existing documents is kept 🚀 

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Create document config in OG document registry - e.g. encounter config
2. Sync to OMS
3. Delete the document registry entry
4. Sync - still exists in OMS (still can create encounters of that type)

## Your environment 🌱

- Open mSupply Version: 2.4
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

**Comment by Chris-Petty** [comment]
So this is because there is no translator for deleting document_registry records, which may have been deliberate...

@lache-melvin @andreievg to stop me going too deep you guys may know the answer - should we be able to delete documents? Surely if so, soft delete?  Otherwise should it cascade the delete to the related form_schema and related patient documents?

**Comment by lache-melvin** [comment]
Yeah I think documents should continue to exist, it's point in time patient info... would soft delete the document registry and leave everything else?

Like you say, form_schema should allow us to delete document registry entries ("current available document types"?) and while keeping past documents around.

Soft delete might serve as a bit more info for us, rather than just orphaned form_schemas? Only gotcha there would be to make sure this works if another patient schema is then added. We only support one patient schema at a time in OMS, would need to make sure still finding the right one

**Comment by Chris-Petty** [comment]
So if someone deleted a patient schema, then added a new one with a new form_schema (as the old one is orphaned) then things would explode? Oh this explains the relation structure I think, document_registry schema changes, then new documents reference that and old document still have their old schema defined?

I guess for patient document registry, soft delete still plays well. We soft delete, if someone wants to make a new one we just undelete the original one and update the `doc_reg.form_schema_id`. Sound good?

What about all the document registry types, any rules I should be considering?
